### DISCUSSION TITLE
Does Roslyn support code fix actions for project/compilation-level diagnostics (with no location)?

### DISCUSSION THREAD

**Discussion by Denis535** [dotnet/roslyn#50087] [Q&A] [answered]
We can report diagnostics for Compilation, SyntaxTree, SyntaxModel but those diagnostics will not have location.

```
private static void OnCompilation(CompilationAnalysisContext context) {
    var compilation = context.Compilation;
    var diagnostic = Diagnostic.Create( Rule, (Location) null );
    context.ReportDiagnostic( diagnostic );
}
```

We can also register code fixes for those diagnostics. But it looks like there is no way to execute those code fixes (there is no light bulb, I can only see it in error list window).

```
public sealed override Task RegisterCodeFixesAsync(CodeFixContext context) {
    var document = context.Document;
    //var span = context.Span;
    var diagnostic = context.Diagnostics.Single();
    RegisterCodeFix( context, "Fix compilation", diagnostic, Action );

    Task<Solution> Action(CancellationToken cancellationToken) {
        return Task.FromResult( document.Project.Solution );
    }
}
```

I even can't test such diagnostic because `CodeFixContext` throws `ArgumentNullException` when document is null.

So, can I make code fixes for a such diagnostics or not?

**Comment by CyrusNajmabadi** [comment, marked as answer]
No. That is not supported.

**Reply by Denis535** [reply]
I don't understand something. 
If `CodeFixProvider` doesn't support project-level diagnostics then why `FixAllContext` has `GetProjectDiagnosticsAsync` method?
